examples/shor.py

```diff
@@ -104,8 +108,10 @@ def high_level_gates(eng, cmd):
 if __name__ == "__main__":
     # build compilation engine list
     resource_counter = ResourceCounter()
-    compilerengines = [AutoReplacer(), InstructionFilter(high_level_gates),
-                       TagRemover(), LocalOptimizer(3), AutoReplacer(),
+    dh = DecompositionRuleSet(modules=[projectq.libs.math,
```

Comment:
I renamed the `dh`'s to `rule_set`, and also renamed a couple files.

I checked that all the examples run (except the ones that require connecting to IBM).